zerver/views/registration.py

```diff
@@ -119,14 +123,21 @@ def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -
     return render(
         request,
         "confirmation/confirm_preregistrationuser.html",
-        context={"key": confirmation_key, "full_name": request.GET.get("full_name", None)},
+        context={"key": confirmation_key, "full_name": full_name},
     )
 
 
 @require_post
-def accounts_register(request: HttpRequest) -> HttpResponse:
+@has_request_variables
+def accounts_register(
+    request: HttpRequest,
+    key: str = REQ(default=""),
+    timezone: str = REQ(default=""),
+    from_confirmation: Optional[str] = REQ(default=None),
+    form_full_name: Optional[str] = REQ("full_name", default=None),
+    raw_source_realm_id: Optional[str] = REQ("source_realm_id", default=None),
```

Comment:
Can this, with some refactoring of the relevant logic below, be changed to use `check_int`?

Reply:
Using `check_int` will invalidate requests that use non-integer realm id. But we can implement a converter that does the same thing for us.